src/vm/memory/segments.zig

```diff
@@ -115,6 +115,20 @@ pub const MemorySegmentManager = struct {
         return relocatable_address;
     }
 
+    // Adds a temporary memory segment and returns the first address of the new segment.
+    pub fn addTempSegment(self: *Self) Relocatable {
+        // Increment the number of temporary segments.
+        self.memory.num_temp_segments += 1;
+
+        // Create the relocatable address for the new segment.
+        const relocatable_address = Relocatable{
+            .segment_index = -@as(i64, @intCast(self.memory.num_temp_segments)),
```

Comment:
Here I think that as the type of `segment_index` in `Relocatable` is known by advance, `.segment_index = -@intCast(self.memory.num_temp_segments)` should be enough no?

Reply:
The `num_temp_segments` is a `u32` by default so this `-@intCast(self.memory.num_temp_segments)` doesn't work in this case.

Reply:
@Godspower-Eze Ah yes true :)